Packs/Fortanix-DSM/Integrations/FortanixDSM/FortanixDSM.py

```diff
@@ -0,0 +1,579 @@
+    state = demisto.args()['state'] if 'state' in demisto.args() else None
+    page = int(demisto.args()['page']) if 'page' in demisto.args() else None
+
+    path = 'crypto/v1/keys'
+    params = {
+        'obj_type': 'SECRET',
+        'sort': 'name:asc'
+    }
+    if state:
+        if state == 'enabled':
+            state = True
+        if state == 'disabled':
+            state = False
+        elif state == 'deleted':
+            params['show_deleted'] = 'true'
+        elif state == 'destroyed':
+            params['show_destroyed'] = 'true'
+
+    if page and page > 1:
+        params['offset'] = str((page - 1) * 100 + 1)
+    if kid:
+        path = path + '/' + kid
+    elif name:
+        params['name'] = name
+    else:
+        if group_id:
+            params['group_id'] = group_id
+        elif GROUP_IDS and len(GROUP_IDS) and GROUP_IDS.find(' ') < 0 and GROUP_IDS.find(',') < 0:
+            params['group_id'] = GROUP_IDS
+
+    res = send_request(path, 'get', params=params)
+    if not res:
+        raise ValueError('Secrets not found')
+
+    if kid and res:
+        readable_output = tableToMarkdown(f'Fortanix DSM Secret {res}', res)
+        return CommandResults(
+            outputs_prefix='Fortanix.Secret',
+            readable_output=readable_output,
+            outputs=res
+        )
+    elif len(res) == 1:
+        readable_output = tableToMarkdown(f'Fortanix DSM Secret {res[0]}', res[0])
+        return CommandResults(
+            outputs_prefix='Fortanix.Secret',
+            readable_output=readable_output,
+            outputs=res[0]
+        )
+    else:
+        mapped_secrets = [{
+            'Name': secret['name'],
+            'ID': secret['kid'],
+            'Group': secret['group_id']
+        } for secret in res if state is None or (state == secret['state'].lower() or state == secret['enabled'])]
+
+        readable_output = tableToMarkdown(f'Found {len(mapped_secrets)} Fortanix DSM Secrets', mapped_secrets)
+        return CommandResults(
+            outputs_prefix='Fortanix.Secret',
+            outputs_key_field='Kid',
+            readable_output=readable_output,
+            outputs=mapped_secrets
+        )
+
+
+def fetch_secret_command():
+
+    if 'kid' in demisto.args():
+        kid = demisto.args()['kid']
+    else:
+        raise ValueError('Secret cannot be fetched without a key ID')
+
+    path = 'crypto/v1/keys/export'
+    body = {
+        'kid': kid
+    }
+
+    res = send_request(path, "post", body)
+    value = ''
+    if not res:
+        raise ValueError('Secret cannot be fetched.')
+    else:
+        try:
+            if 'value' in res:
+                if isBase64(res['value']):
+                    value = b64decode(res['value']).decode("utf-8")
+                else:
+                    value = res['value']
+            else:
+                value = res
+        except Exception:
+            value = res['value']
```

Comment:
Else statement can be removed, if result is empty it will raise an exception and quit the function.